run_common.go

```diff
@@ -2092,3 +2092,28 @@ func relabel(path, mountLabel string, shared bool) error {
 	}
 	return nil
 }
+
+// mapContainerNameToHostname returns the passed-in string with characters that
+// don't match the pattern "[A-Za-z0-9][A-Za-z0-9._-]+" stripped out.
+func mapContainerNameToHostname(containerName string) string {
+	const initialAllowedChars = "abcdefghijklmnopqrstuvwxyz" + "ABCDEFGHIJKLMNOPQRSTUVWXYZ" + "0123456789"
+	const allowedChars = initialAllowedChars + "._-"
```

Comment:
`_` is not a valid hostname char and must be stripped, the same is done in podman https://github.com/containers/podman/pull/24675/commits/6fa234a5de44b5279b244e06e889e74d9d27ae40